backend/app/services/storage.py

```
@@ -0,0 +1,430 @@
+
+try:
+    use_emulator = settings.use_firebase_emulator
+    service_account_path = settings.firebase_service_account_path
+    project_id = settings.firebase_project_id
+
+    if use_emulator:
+        os.environ["FIREBASE_STORAGE_EMULATOR_HOST"] = "http://127.0.0.1:4000"
+        logger.info("Using Firebase Storage Emulator")
+
+    if not project_id:
+        raise ValueError("FIREBASE_PROJECT_ID environment variable is not set.")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)